src/main/java/duke/command/AddTaskCommand.java

```
@@ -0,0 +1,80 @@
+
+        switch(instruction) {
+            case "todo":
+                task = new ToDo(description);
+                tasks.addTask(task);
+                return ui.addTask(task, tasks.getNumber());
+
+            case "event":
+
+                StringBuilder at = new StringBuilder();
+                for (int i = 0; i < words.length; i++) {
+                    String word = words[i];
+                    if (!word.equals("/at")) {
+                        descriptions.append(words[i]);
+                        descriptions.append(" ");
+                    } else {
+                        for (int j = i + 1; j < words.length; j++) {
+                            at.append(words[j]);
+                            at.append(" ");
+                        }
+                        break;
+                    }
+                }
+                task = new Event(descriptions.toString().trim(), at.toString().trim());
+                tasks.addTask(task);
+                return ui.addTask(task, tasks.getNumber());
+
+            case "deadline":
+                StringBuilder by = new StringBuilder();
+                for (int i = 0; i < words.length; i++) {
+                    String word = words[i];
+                    if (!word.equals("/by")) {
+                        descriptions.append(words[i]);
+                        descriptions.append(" ");
+                    } else {
+                        for (int j = i + 1; j < words.length; j++) {
+                            by.append(words[j]);
+                            by.append(" ");
+                        }
+                        break;
+                    }
+                }
+                task = new Deadline(descriptions.toString().trim(), by.toString().trim());
+                tasks.addTask(task);
+                return ui.addTask(task, tasks.getNumber());
+
+            default:
+                return "This shouldn't appear.";
```

Perhaps you'd want to throw an exception here instead

---

Okay thanks!